src/compiler/interface.jl

```diff
@@ -122,13 +122,24 @@ function copy!(x::AbstractVector, ps::Params)
   ps
 end
 
+"""
+    gradient(() -> loss(), ::Params) -> Grads
+
+Gradient with implicit parameters. Returns a container, from which
+`grads[W]` extracts the gradient with respect to `W`.
+This contains as `grads.value` the value of the function.
```

Comment:
`value` referred to gradient can be confusing. Can we think of a better name? `output`? 
or maybe is fine already

Reply:
Could be `output`, or `primal`, is that an official term or one we made up in Julia? Maybe `forward` is another option?

`value` is what ForwardDiff uses, but I guess it's usage is in fact a little different. I mean the elements apart from the value are its partials, numbers of the same units/shape/etc. While the other parts here are the actual gradients (and params), completely different objects.